### PR TITLE
Task 2 - linter and prettier

### LINKED ISSUE #4: Add Eslint and Prettier
Add EsLint and Prettier configuration for better practices

### PR DESCRIPTION
**Pull request by mugar97**
Adding Eslint and prettier. 

Resolves #4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,14 +13,16 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-scripts": "5.0.1",
-    "typescript": "^4.9.5",
```
Comment (epam-kursikov) [inline package.json:16]: Why did you delete it?

File: package.json [hunk 2]
```diff
@@ -13,14 +13,16 @@
...
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "lint": "eslint src/**/*.{ts,tsx}",
+    "lint:fix": "eslint --fix 'src/**/*.{ts,tsx}'",
+    "format": "prettier --write 'src/**/*.{ts,tsx,css}' --config ./.prettierrc"
```
Comment (epam-kursikov) [inline package.json]: You might not need to specify config explicitly since `.prettierrc` is the default path. Not sure though.

File: src/App.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { ChangeEvent, FormEvent } from "react";
+import React, { type ReactElement, type ChangeEvent, type FormEvent } from 'react';
```
Comment (epam-kursikov) [inline src/App.tsx]: Why did you add `type` for type imports?

File: src/App.tsx [hunk 4]
```diff
@@ -20,21 +20,21 @@ class Counter extends React.Component<CounterProps> {
...
     this.setState({ value: this.state.value + 1 });
   }
 
-  decrement() {
+  decrement(): void {
     this.setState({ value: this.state.value - 1 });
   }
 
-  render() {
+  render(): ReactElement {
```
Comment (epam-kursikov) [inline src/App.tsx]: I'd say this is too much typing. Typing is great but this is overzealous :)
If your lint config forces you to type absolutely everything then it's a bad config. If this was your own decision then you can go for it but I see no benefit in such obvious types.

File: package.json (hunk 1 above)
Comment (mugar97) [reply package.json:16]: It was moved as a dev dependency at some point, is typescript considered a prod library? or dev? 🤔

File: src/App.tsx (hunk 3 above)
Comment (mugar97) [reply src/App.tsx]: It was suggested by eslint for this Rule: https://typescript-eslint.io/rules/consistent-type-imports

File: package.json (hunk 1 above)
Comment (epam-kursikov) [reply package.json:16]: It's a dev dependency. I asked because I didn't see it added to `devDependencies`.

File: src/App.tsx (hunk 3 above)
Comment (epam-kursikov) [reply src/App.tsx]: I see, nothing wrong here just unusual to me.

File: package.json (hunk 2 above)
Comment (mugar97) [reply package.json]: updated!

File: src/App.tsx (hunk 4 above)
Comment (mugar97) [reply src/App.tsx]: I removed all the unusual suggestions, including this one. thank you

**Review by epam-kursikov — APPROVED** [review]
Lok good. Never hesitate to tweak linter's config -- it should help you, not be in your way.
